internal/adapters/dataproviders/database/mongo/pegin.go

```diff
@@ -208,3 +210,60 @@ func (repo *peginMongoRepository) DeleteQuotes(ctx context.Context, quotes []str
 	}
 	return uint(peginResult.DeletedCount + retainedResult.DeletedCount + creationDataResult.DeletedCount), nil
 }
+
+func (repo *peginMongoRepository) ListQuotesByDateRange(ctx context.Context, startDate, endDate time.Time) ([]quote.PeginQuote, []quote.RetainedPeginQuote, error) {
```

Comment:
you can find the relationship between the quote and the retained quote in here, this would remove the need of querying again in the use case, you just need to return a struct with the quote and retained to maintain the relationship once you leave this layer